---
layout: post
title: What's going on here...
---

Guess I should actually post something here now that I might have some content worth sharing. 

<h1>Oregon Is Cool I Guess</h1>


<img src="https://i.imgur.com/K2zC9Gz.jpg" align="left" width="200">Been trying to do some traveling when I have free time. I went on a solo trip to Portland almost as soon as I moved here. Most of my trip was spent in Washington Park, just taking the trails and exploring. 
The International Test Rose Gardens there were just as beautiful as I imagined them to be. Got to eat some good food there too, Fried Egg I'm in Love definitely lives up to the hype. Another good place is The Sports Bra, super awesome atmosphere and good bar food. 
Finally, I got to go to a super good tea place called the Tao of Tea. The city itself is probably the grungiest place I've ever been but I love just walking around and looking at things. <br> I've also gone on a couple excellent trips to the coast, 
something that was obviously impossible for me back home. It's so beautiful and takes my breath away every time. My dog loves to drink salt water. I've also explored Salem, Newport, and done a little bit of hiking. Trips in the future will probably be to Tillamook, Eugene, and Bend.<br>

<h1>Isolated chloroplast genes + Lewisia phylogeny</h1>

Alright, let's get to the boring stuff. My samples came back for my grad research Halloween day so I ran through the chloroplast DNA workflow on it. It was pretty simple to get my head around, but I'm hoping that explaining it will further cement my knowledge on the subject.

The chloroplast genome is very similar across plant life. It is prokaryotic in origin and is inherited maternally. They tend to be around 150000 BP (in fact, <a href="https://i.imgur.com/8a0Aleo.png">the ones I looked at were all *basically* 151411 bp</a>) and look like <a href="https://i.imgur.com/IfN9opC.png">this</A>.
Anyways, they're pretty easy to ID in the context of a full genome since they have a very recognizable pattern - especially with the two IR'S (inverted repeats).<br>

So! I take my raw sequence data and run it through BBduk (<a href="https://jgi.doe.gov/data-and-tools/software-tools/bbtools/">a tool in the toolbox of BBtools</a>) which trims adapters. I get a bunch of .fq files, which I can then use <a href="https://github.com/Kinggerm/GetOrganelle">GetOrganelle </a> to ID the chloroplast genome in the sequences.
After this, it's simply a matter of aligning the sequences with <a href="https://github.com/lh3/minimap2">MiniMap 2</a> and then visualizing the relationships by constructing a phylogenetic tree (which my PI did for me :3c). Now that the chloroplast genome has been ID'd, the project is now handed over to me for TE identification. I'm only a little terrified.
<br><br>
My other weekend project was super cool, I'm co-authoring the section on Lewisia for OregonFlora, a plant ID book that's about to release its third book. My PI suggested I download some GenBank sequences and create a phylogenetic tree. Easy, right?
<br><br>
A very crucial element is as follows: I am a dumbass. Anyways.<br><br>

So, I start looking around at tools to download sequences from GenBank. NCBI has a great set of <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/download-and-install/">database download tools</a> - I could NOT figure out how to use them. So, I take the easy way and use <a href="https://www.ncbi.nlm.nih.gov/books/NBK25501/">E-Utils</a>.
My command is as follows below (after install E-Utils, obvs).<br><br>

<img src="https://i.imgur.com/pFBzQOH.png"><br><br>

Okay, cool, there's some sequences in there and they're seperated by accession. No idea what's going on but I decide to follow the documentation for <a href="https://sites.google.com/view/cd-hit">CD-HIT</a> and cluster these sequences.<br><br>

Yep, that was a lie. I forgot to cluster my sequences the first time, ran through the ENTIRE workflow, and ended up with this absolute beast.<br>

<img src="https://i.imgur.com/rthVgCT.png"><br><br>

Well, we could talk about my failures all day but let's continue. My PI sets me straight, I cluster the sequences then align them with <a href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a> (super cool tool!). <a href="http://www.iqtree.org/">IQ-TREE</a> is used to construct the phylogenetic tree, and I throw the aligned sequences into <a href="https://www.jalview.org/">JalView</a> to see what's going on.<br><br> 

<img src="https://i.imgur.com/HSnr0yv.png"><br><br>

I think I still have it a little messed up, I think that there's some sequences clustered where they shouldn't be. Oh well. It's a learning process.<br><br>

I'm currently in the process of setting up a Raspberry Pi as a home server, this will probably be one of the subjects of my next post. I'm really struggling with this as my computer is REFUSING to connect to the pi over SSH.

<h1>Beers</h1>

I also drank a lot of beers. And some other stuff. Yeah the formatting is messed up I don't know how to fix it.<br>

<img src="https://i.imgur.com/yBp95By.jpg" align="left" width="200"><br>

<img src="https://i.imgur.com/DEvCbfI.jpg" align ="right" width="200"><br>

<img src="https://i.imgur.com/MRh1S7d.jpg" align="left" width="200"><br>

<img src="https://i.imgur.com/M7yxnBv.jpg" align="right" width="200"><br>

<img src="https://i.imgur.com/ouOyuPo.jpg" align="left" width="200"><br>
